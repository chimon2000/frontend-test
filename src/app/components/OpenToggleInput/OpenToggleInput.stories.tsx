import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { action } from "@storybook/addon-actions";
import { withKnobs, boolean } from "@storybook/addon-knobs";

import { OpenToggleInput } from "./OpenToggleInput";

const stories = storiesOf("Components", module);

stories.add(
  "Open Toggle Input",
  withInfo({ inline: true })(() => (
    <OpenToggleInput
      name={"toggle-sample"}
      id={"toggle_sample"}
      checked={boolean("Checked", false)}
      onChange={action("Changed")}
    />
  ))
);
