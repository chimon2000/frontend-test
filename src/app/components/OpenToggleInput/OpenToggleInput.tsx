import * as React from "react";

import Checkbox, { ChangeEventObject } from "../Checkbox";

export interface OpenToggleInputProps {
  className?: string;
  checked: boolean;
  name: string;
  id: string;
  onChange: (result: ChangeEventObject) => void;
}

export const OpenToggleInput: React.FunctionComponent<OpenToggleInputProps> = ({
  checked,
  onChange,
  name,
  id
}) => (
  <div className="open-toggle-input">
    <Checkbox
      label="Open Now"
      checked={checked}
      className="checkbox--radio"
      name={name}
      id={id}
      onChange={onChange}
    />
  </div>
);
