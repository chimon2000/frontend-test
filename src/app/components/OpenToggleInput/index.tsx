export { OpenToggleInput as default } from "./OpenToggleInput";
