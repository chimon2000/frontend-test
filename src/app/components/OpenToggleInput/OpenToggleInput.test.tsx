import * as React from "react";
import { shallow } from "enzyme";

import { OpenToggleInput } from "./OpenToggleInput";

describe("<OpenToggleInput />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(
      <OpenToggleInput
        name={"toggle-sample"}
        id={"toggle_sample"}
        checked={false}
        onChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
