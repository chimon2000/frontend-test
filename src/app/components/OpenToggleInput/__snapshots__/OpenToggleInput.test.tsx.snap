// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpenToggleInput /> doesn't blow up 1`] = `
<div
  className="open-toggle-input"
>
  <Component
    checked={false}
    className="checkbox--radio"
    id="toggle_sample"
    label="Open Now"
    name="toggle-sample"
    onChange={[Function]}
  />
</div>
`;
